packages/web-react/src/components/Stack/Stack.tsx

```diff
@@ -1,30 +1,24 @@
-import React, { ElementType, JSXElementConstructor } from 'react';
+import React, { ElementType } from 'react';
 import classNames from 'classnames';
-import { useStyleProps } from '../../hooks/styleProps';
-import { useClassNamePrefix } from '../../hooks/useClassNamePrefix';
-import { ChildrenProps, StyleProps, TransferProps } from '../../types';
-
-export interface StackProps<T extends ElementType = 'div'> extends ChildrenProps, StyleProps, TransferProps {
-  /**
-   * The HTML element or React element used to render the Stack, e.g. 'div', 'ul', or something else.
-   *
-   * @default 'div'
-   */
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  elementType?: T | JSXElementConstructor<any>;
-}
+import { useStackStyleProps } from './useStackStyleProps';
+import { useStyleProps } from '../../hooks';
+import { SpiritStackProps, StyleProps } from '../../types';
 
 const defaultProps = {
   elementType: 'div',
+  hasSpacing: true,
+  hasBottomDivider: false,
+  hasMiddleDividers: false,
+  hasTopDivider: false,
 };
 
-export const Stack = <T extends ElementType = 'div'>(props: StackProps<T>): JSX.Element => {
+export const Stack = <T extends ElementType = 'div'>(props: SpiritStackProps<T>): JSX.Element => {
   const { elementType: ElementTag = 'div', children, ...restProps } = props;
-  const stackClass = useClassNamePrefix('Stack');
-  const { styleProps, props: otherProps } = useStyleProps(restProps);
+  const { classProps, props: modifiedProps } = useStackStyleProps(restProps);
+  const { styleProps, props: otherProps } = useStyleProps(modifiedProps as StyleProps);
```

Comment:
I do not like the `as StyleProps`. This suggests that is some inconsistency between returned and inputted values.